pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1229,56 +1239,76 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 		metrics.DurationAppliedToGroupSyncing.Observe(float64(d.Milliseconds()))
 		klog.V(2).Infof("Finished syncing AppliedToGroup %s. (%v)", key, d)
 	}()
-	var pods []*v1.Pod
 	appGroupNodeNames := sets.String{}
 	appliedToGroupObj, found, _ := n.appliedToGroupStore.Get(key)
 	if !found {
 		klog.V(2).Infof("AppliedToGroup %s not found.", key)
 		return nil
 	}
-	memberSetByNode := make(map[string]controlplane.GroupMemberSet)
-	scheduledPodNum, scheduledExtEntityNum := 0, 0
 	appliedToGroup := appliedToGroupObj.(*antreatypes.AppliedToGroup)
-	pods, externalEntities := n.getAppliedToWorkloads(appliedToGroup)
-	for _, pod := range pods {
-		if pod.Spec.NodeName == "" || pod.Spec.HostNetwork == true {
-			// No need to process Pod when it's not scheduled.
-			// HostNetwork Pods will not be applied to by policies.
-			continue
+	memberSetByNode := make(map[string]controlplane.GroupMemberSet)
+	var updatedAppliedToGroup *antreatypes.AppliedToGroup
+	if appliedToGroup.Service != nil {
+		// AppliedToGroup for NodePort Service span to all Nodes.
+		nodeList, err := n.kubeClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
it should *not* use kubeClient to list Nodes from kube-apiserver every time it handles this group. The `nodeLister` is already available in this controller, which has cached all Nodes.

Reply:
Ah my bad. Changed.